### PR TITLE
Update `ethereumjs-tx` to `@ethereumjs/tx` to support EIP1559 transactions

### PR DESCRIPTION
**Pull request by rickycodes**
We need this in the mobile app to interoperate with type two (eip1559) transactions without crashing the app

The fake transaction re-implementation is 1:1 from https://github.com/ethereumjs/ethereumjs-monorepo/tree/master/packages/tx#fake-transaction

re: https://github.com/MetaMask/metamask-mobile/issues/2851

### CODE REVIEW

File: subproviders/hooked-wallet-ethtx.js [hunk 1]
```diff
@@ -32,7 +32,7 @@ function HookedWalletEthTxSubprovider(opts) {
     opts.getPrivateKey(txData.from, function(err, privateKey) {
       if (err) return cb(err)
 
-      var tx = new EthTx(txData)
+      var tx = new Transaction(txData)
```
Comment (brad-decker) [inline subproviders/hooked-wallet-ethtx.js:35]: why is this not TransactionData.fromTxData?

File: subproviders/vm.js [hunk 2]
```diff
@@ -143,11 +142,13 @@ VmSubprovider.prototype.runVm = function(payload, cb){
     gasPrice: txParams.gasPrice ? ethUtil.addHexPrefix(txParams.gasPrice) : undefined,
     nonce: txParams.nonce ? ethUtil.addHexPrefix(txParams.nonce) : undefined,
   }
-  var tx = new FakeTransaction(normalizedTxParams)
-  tx._from = normalizedTxParams.from || '0x0000000000000000000000000000000000000000'
+  const tx = new Transaction(normalizedTxParams)
```
Comment (brad-decker) [inline subproviders/vm.js:145]: why not fromTxData?

**Review by brad-decker — CHANGES_REQUESTED** [review]
questions around the instantiation method.

File: subproviders/hooked-wallet-ethtx.js (hunk 1 above)
Comment (rickycodes) [reply subproviders/hooked-wallet-ethtx.js]: i just looked at it would appear: no reason :sweat_smile:

**Review by brad-decker — APPROVED** [review]

**Pull request merged**